src/test/java/org/jitsi/meet/test/DesktopSharingTest.java

```diff
@@ -167,20 +116,69 @@ public void testDesktopSharingInPresence()
         catch (TimeoutException e)
         {
             assertEquals(
-                expectedResult, remoteVideoType[0],
-                "Wrong video type, maybe desktop sharing didn't work");
+                    expectedResult, remoteVideoType[0],
+                    "Wrong video type, maybe desktop sharing didn't work");
         }
+    }
 
-        // allow the participant to leave
-        try
-        {
-            waitEndSignal.await(10, TimeUnit.SECONDS);
-            print("End DesktopSharingTest.");
-        }
-        catch (InterruptedException e)
-        {
-            e.printStackTrace();
-        }
-        ensureTwoParticipants();
+    /**
+     * Starts desktop sharing.
+     */
+    private void startDesktopSharing()
+    {
+        TestUtils.waitForElementByXPath(
```

Comment:
So this 3 statements in the PageObject world would be:
Toolbat toolbar = getParticipant2().getToolbar();
Button dsButton = toolbar.getDesktopsharingButton();
dsButton.click();
dsButton.assertToggled(true /* toggled */, 2 /* timeout seconds */);
I don't know what would be the exact type of the Button. Maybe it's just WebElement. Or maybe we need a class the wraps it.

Reply:
I think the approach with getButton and then click is not good at all in the context of react. I would even say to skip searching for web element and getting them and then executing click. We already had such problems with the toolbars. The better approach is at the end to just execute document.getElementById('...').click();
https://github.com/jitsi/jitsi-meet-torture/blob/master/src/test/java/org/jitsi/meet/test/util/MeetUIUtils.java#L243
The problem is a StaleElementException coming from selenium. You take the element and try to click it, but by that time react has re-rendered the toolbar and that element is stale.

Reply:
Note that this is a problem with saving the WebElement, and not a problem with the "getButton" abstraction itself.

Reply:
Yep, that's right, thanks for the clarification :)